spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestSnapshotSelection.java

```diff
@@ -425,16 +426,35 @@ public void testSnapshotSelectionByBranchWithSchemaChange() throws IOException {
     // Deleting a column to indicate schema change
     table.updateSchema().deleteColumn("data").commit();
 
-    // The data should have the deleted column as it was captured in an earlier snapshot.
-    Dataset<Row> deletedColumnBranchSnapshotResult =
-        spark.read().format("iceberg").option("branch", "branch").load(tableLocation);
-    List<SimpleRecord> deletedColumnBranchSnapshotRecords =
-        deletedColumnBranchSnapshotResult
-            .orderBy("id")
-            .as(Encoders.bean(SimpleRecord.class))
-            .collectAsList();
-    Assert.assertEquals(
-        "Current snapshot rows should match", expectedRecords, deletedColumnBranchSnapshotRecords);
+    // The data should not have the deleted column
+    Assertions.assertThat(
+            spark
+                .read()
+                .format("iceberg")
+                .option("branch", "branch")
+                .load(tableLocation)
+                .orderBy("id")
+                .collectAsList())
+        .containsExactly(
+            new GenericRowWithSchema(new Object[] {1}, null),
+            new GenericRowWithSchema(new Object[] {2}, null),
+            new GenericRowWithSchema(new Object[] {3}, null));
```

Comment:
Can you use SimpleRecord like the rest of the tests do instead?

Reply:
unfortunately that doesn't work, because `SimpleRecord` expects the `data` field to be populated